### DISCUSSION TITLE
Load image from ARGB bytearray

### DISCUSSION THREAD

**Discussion by feliwir** [SixLabors/ImageSharp#1344] [Q&A] [answered]
Hello,

i'm trying to load an `Image<Bgra32>` from a given byte array with a given width & height. However the `Image.Load<Bgra32>` function only takes a byte array, so i'm not sure how to my width & height in there. So what i'm trying to do is this:
```csharp
var image = new Image<Argb32>(width, height,pixelData);
```

Probably this is really easy, but i can't find a method on how to do this. 
Help is appreciated, thanks!

**Comment by tocsoft** [comment, marked as answer]
You are looking for the `Image.LoadPixelData<Bgra32>(data, width, hieght)` static method.

**Reply by JimBobSquarePants** [reply]
https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Image.html#SixLabors_ImageSharp_Image_LoadPixelData__1___0___System_Int32_System_Int32_

**Reply by feliwir** [reply]
Thanks!

**Comment by feliwir** [comment]
Sorry to write here again, but i just realized my pixeldata is in ABGR32 format, which doesn't seem to exist natively in ImageSharp. Do i need to manually convert the data?

**Reply by JimBobSquarePants** [reply]
Or create your own pixel format. It won't have the various optimizations in place for RGBA compatible types but would still work. 

What's the source of this data? It's not a format I've come across in the wild, if it's common enough we could simply add it to the library.

**Reply by feliwir** [reply]
@JimBobSquarePants we're loading `.ani`  files (animated windows cursors), but it turned out the pixelformat was **BGRA** in the end, so it was my bad. I was just confused because SDL2 stored this format as `SDL_PIXELFORMAT_ABGR8888`.

If you're interested in that format you can take a look at our code from here:
https://github.com/OpenSAGE/OpenSAGE/tree/master/src/OpenSage.Game/Data/Ani

**Reply by antonfirsov** [reply]
@feliwir sounds reasonable, opened #1345

**Reply by antonfirsov** [reply]
@feliwir off question:
what is the difference between `SDL_PIXELFORMAT_BGR888` and `SDL_PIXELFORMAT_BGR24`?

**Reply by feliwir** [reply]
To clarify this a bit: 
**SDL_PIXELFORMAT_ABGR8888** is indentical to **Bgra24** in ImageSharp. So i don't really need #1345 for myself anymore. Not sure why the SDL2 naming convention is different from ImageSharp in that regard

> @feliwir off question:
> what is the difference between `SDL_PIXELFORMAT_BGR888` and `SDL_PIXELFORMAT_BGR24`?

According to the [docs](https://wiki.libsdl.org/SDL_PixelFormatEnum) it's just an alias.
